### PR TITLE
docs: Switch to flat config by default

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Switched the default configuration documentation to refer to flat config instead of eslintrc. Primarily, this involved taking all of the content on `configuration-files-new.md` and integrating it into the multi-page configuration documentation we already had. This required moving things around and removing duplicate content. 

Refs #13481

#### Is there anything you'd like reviewers to focus on?

Please double-check that I haven't left duplicate content in the files and that the structure makes sense. Everything isn't exactly 1-to-1 with eslintrc, so I did my best.

### CODE REVIEW

File: docs/src/use/configure/configuration-files-new.md
Comment (bmish) [inline docs/src/use/configure/configuration-files-new.md:1]: What do you think of keeping the file with a link to the replacement to avoid breaking links? Similar to this discussion: https://github.com/eslint/eslint/pull/17792#discussion_r1405457869

Comment (bmish) [reply docs/src/use/configure/configuration-files-new.md:1]: Either a link or redirect to the replacement is fine.

Comment (nzakas) [reply docs/src/use/configure/configuration-files-new.md:1]: Yeah, we'll do a redirect, but that needs to be done in the eslint.org repo.

Comment (nzakas) [reply docs/src/use/configure/configuration-files-new.md:1]: https://github.com/eslint/eslint.org/pull/499

File: docs/src/use/configure/rules.md [hunk 1]
```diff
@@ -69,76 +73,96 @@ Configuration comments can include descriptions to explain why the comment is ne
...
+```js
+export default [
+    {
+        rules: {
+            eqeqeq: "off",
+            "no-unused-vars": "error",
+            "prefer-const": "warn"
+        }
+
     }
```
Comment (kecrily) [inline docs/src/use/configure/rules.md:89]:
```suggestion
        }
    }
```

File: docs/src/use/configure/rules-deprecated.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+To report unused `eslint-disable` comments, use the `reportUnusedDisableDirectives` setting. For example:
+
+```json
+{
+  "rules": {...},
+  "reportUnusedDisableDirectives": true
+}
+```
+
+This setting is similar to [--report-unused-disable-directives](../command-line-interface#--report-unused-disable-directives) CLI option, but doesn't fail linting (reports as `"warn"` severity).
```
Comment (kecrily) [inline docs/src/use/configure/rules-deprecated.md:318]:
```suggestion
This setting is similar to [`--report-unused-disable-directives`](../command-line-interface#--report-unused-disable-directives) CLI option, but doesn't fail linting (reports as `"warn"` severity).
```

File: docs/src/use/configure/rules-deprecated.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+To disable all inline config comments, use the `noInlineConfig` setting in your configuration file. For example:
+
+```json
+{
+  "rules": {...},
+  "noInlineConfig": true
+}
+```
+
+You can also use the [--no-inline-config](../command-line-interface#--no-inline-config) CLI option to disable rule comments, in addition to other in-line configuration.
```
Comment (kecrily) [inline docs/src/use/configure/rules-deprecated.md:305]:
```suggestion
You can also use the [`--no-inline-config`](../command-line-interface#--no-inline-config) CLI option to disable rule comments, in addition to other in-line configuration.
```

File: docs/src/use/configure/rules.md [hunk 4]
```diff
@@ -277,31 +305,47 @@ console.log('hello');
...
+    {
+      files: ["*-test.js","*.spec.js"],
+      rules: {
+        "no-unused-expressions": "off"
+      }
+    }
+];
 ```
 
 You can also use the [--no-inline-config](../command-line-interface#--no-inline-config) CLI option to disable rule comments, in addition to other in-line configuration.
```
Comment (kecrily) [inline docs/src/use/configure/rules.md:351]:
```suggestion
You can also use the [`--no-inline-config`](../command-line-interface#--no-inline-config) CLI option to disable rule comments, in addition to other in-line configuration.
```

File: docs/src/use/configure/rules.md [hunk 5]
```diff
@@ -310,11 +354,15 @@ You can also use the [--no-inline-config](../command-line-interface#--no-inline-
...
+export default [
+    {
+        linterOptions: {
+            reportUnusedDisableDirectives: true
+        }
+    }
+];
 ```
 
 This setting is similar to [--report-unused-disable-directives](../command-line-interface#--report-unused-disable-directives) and [--report-unused-disable-directives-severity](../command-line-interface#--report-unused-disable-directives-severity) CLI options.
```
Comment (kecrily) [inline docs/src/use/configure/rules.md:368]:
```suggestion
This setting is similar to [`--report-unused-disable-directives`](../command-line-interface#--report-unused-disable-directives) and [`--report-unused-disable-directives-severity`](../command-line-interface#--report-unused-disable-directives-severity) CLI options.
```

File: docs/src/use/configure/ignore.md [hunk 6]
```diff
@@ -8,135 +8,76 @@ eleventyNavigation:
...
+```js
+// eslint.config.js
+export default [
+    {
+        ignores: ["foo.js"]
+    }
+]
 ```
 
 And then you run:
```
Comment (mdjermanovic) [inline docs/src/use/configure/ignore.md:83]: For consistency with other examples, it might be good to use `npx` in this one too (`eslint foo.js` -> `npx eslint foo.js` on line 86; GitHub doesn't let me leave a suggestion there).

File: docs/src/use/configure/ignore.md [hunk 7]
```diff
@@ -8,135 +8,76 @@ eleventyNavigation:
 
 ---
```
Comment (mdjermanovic) [inline docs/src/use/configure/ignore.md:9]:
```suggestion
---

::: tip
This page explains how to ignore files using the flat config format. For the deprecated eslintrc format, [see the deprecated documentation](ignore-deprecated).
:::
```

Since other documents have this tip, I'm guessing this one should have it too.

File: docs/src/use/configure/configuration-files-deprecated.md [hunk 8]
```diff
@@ -0,0 +1,445 @@
+---
+title: Configuration Files
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files-deprecated.md:2]:
```suggestion
title: Configuration Files (Deprecated)
```

File: docs/src/use/configure/language-options-deprecated.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
+---
+title: Configure Language Options
```
Comment (mdjermanovic) [inline docs/src/use/configure/language-options-deprecated.md:2]:
```suggestion
title: Configure Language Options (Deprecated)
```

File: docs/src/use/configure/configuration-files-new.md
Comment (mdjermanovic) [reply docs/src/use/configure/configuration-files-new.md:1]: Since we are going to redirect, perhaps we should remove the `configuration-files-new.md` file now? It appears in the navigation:

image

File: docs/src/use/getting-started.md [hunk 10]
```diff
@@ -103,34 +114,32 @@ You can also manually set up ESLint in your project.
...
-   }
+   import js from "@eslint/js";
+
+   export default [
+       js.configs.recommended,
+
+       rules: {
+           "no-unused-vars": "warn",
+           "no-undef": "warn"
+       }
```
Comment (mdjermanovic) [inline docs/src/use/getting-started.md:141]:
```suggestion
      {
          rules: {
              "no-unused-vars": "warn",
              "no-undef": "warn"
          }
      }
```

File: docs/src/use/configure/parser.md [hunk 11]
```diff
@@ -7,32 +7,67 @@ eleventyNavigation:
...
-
-1. It must be a Node module loadable from the config file where the parser is used. Usually, this means you should install the parser package separately using npm.
-1. It must conform to the [parser interface](../../extend/custom-parsers).
-
-Note that even with these compatibilities, there are no guarantees that an external parser works correctly with ESLint. ESLint does not fix bugs related to incompatibilities with other parsers.
+In many cases, you can use the [default parser](https://github.com/eslint/espree) that ESLint ships with for parsing your JavaScript code. You can optionally override the default parser by using the `parser` property. The `parser` property must be an object containing either a `parse()` method or a `parseForESLint()` method (read about the [parser interface](../../extend/custom-parsers)). For example, you can use the [`@babel/eslint-parser`](https://www.npmjs.com/package/@babel/eslint-parser) package to allow ESLint to parse experimental syntax:
```
Comment (mdjermanovic) [inline docs/src/use/configure/parser.md:18]:
```suggestion
In many cases, you can use the [default parser](https://github.com/eslint/espree) that ESLint ships with for parsing your JavaScript code. You can optionally override the default parser by using the `parser` property. The `parser` property must be an object that conforms to the [parser interface](../../extend/custom-parsers). For example, you can use the [`@babel/eslint-parser`](https://www.npmjs.com/package/@babel/eslint-parser) package to allow ESLint to parse experimental syntax:
```

Since this is a user guide, I think it isn't necessary to mention technical details like `parse()`/`parseForESLint()` but rather just link to the required interface specification and thus also imply that not every parser can be used with ESLint.

File: docs/src/use/configure/parser.md [hunk 12]
```diff
@@ -7,32 +7,67 @@ eleventyNavigation:
...
+    {
+        languageOptions: {
+            parser: babelParser,
+            parserOptions: {
+                requireConfigFile: false,
+                babelOptions: {
+                  babelrc: false,
+                  configFile: false,
+                  presets: ["@babel/preset-env"],
+                },
```
Comment (mdjermanovic) [inline docs/src/use/configure/parser.md:64]:
```suggestion
                  presets: ["@babel/preset-env"]
                }
```

File: docs/src/use/configure/language-options.md [hunk 13]
```diff
@@ -5,129 +5,67 @@ eleventyNavigation:
...
-        "env": {
-            "example/custom": true
+```js
+// eslint.config.js
+export default [
+    {
+        languageOptions: {
+            ecmaFeatures: {
+                jsx: true
+            }
```
Comment (mdjermanovic) [inline docs/src/use/configure/language-options.md:59]:
```suggestion
            parserOptions: {
                ecmaFeatures: {
                    jsx: true
                }
            }
```

File: docs/src/use/configure/language-options.md [hunk 14]
```diff
@@ -5,129 +5,67 @@ eleventyNavigation:
...
-    browser: true
-    node: true
+];
 ```
 
-### Using a plugin
+## Specifying Parser Options
 
-If you want to use an environment from a plugin, be sure to specify the plugin name in the `plugins` array and then use the unprefixed plugin name, followed by a slash, followed by the environment name. For example:
+If you are using the built-in ESLint parser, you can additionally change how ESLint interprets your code by specifying the `languageOptions.parserOptions` key. All language options are `false` by default:
```
Comment (mdjermanovic) [inline docs/src/use/configure/language-options.md:42]:
```suggestion
If you are using the built-in ESLint parser, you can additionally change how ESLint interprets your code by specifying the `languageOptions.parserOptions` key. All options are `false` by default:
```

This may sound like it refers to `languageOptions` properties, while it refers to the default parser's options.

File: docs/src/use/configure/plugins.md [hunk 15]
```diff
@@ -8,6 +8,10 @@ eleventyNavigation:
 
 ---
 
+::: tip
+This page explains how to configure plugins using the flat config format. For the deprecated eslintrc format, [see the deprecated documentation](plugins-deprecated).
+:::
+
 You can extend ESLint with plugins in a variety of different ways. Plugins can include:
 
 * Custom rules to validate if your code meets a certain expectation, and what to do if it does not meet that expectation.
```
[the remark this answers is not in the document]
Comment (mdjermanovic) [reply docs/src/use/configure/plugins.md:17]: Also, there's "Custom configurations" in this list, but the rest of the document doesn't mention it. Perhaps we could add a short section that says something like "Plugins can export configurations. For the usage, please refer to plugin's documentation."

File: docs/src/use/configure/rules.md [hunk 16]
```diff
@@ -69,76 +73,96 @@ Configuration comments can include descriptions to explain why the comment is ne
...
-        "eqeqeq": "off",
-        "curly": "error",
-        "quotes": ["error", "double"]
+```js
+export default [
+    {
+        rules: {
+            eqeqeq: "off",
+            "no-unused-vars": "error",
+            "prefer-const": "warn"
```
Comment (mdjermanovic) [inline docs/src/use/configure/rules.md:86]:
```suggestion
            "prefer-const": ["error", { "ignoreReadBeforeAssign": true }]
```

To have an example with options, since the sentence before this code block mentions options.

File: docs/src/use/configure/rules.md [hunk 17]
```diff
@@ -258,7 +284,9 @@ foo(); // eslint-disable-line example/rule-name
 foo(); /* eslint-disable-line example/rule-name */
 ```
 
-**Note:** Comments that disable warnings for a portion of a file tell ESLint not to report rule violations for the disabled code. ESLint still parses the entire file, however, so disabled code still needs to be syntactically valid JavaScript.
+::: note
```
Comment (mdjermanovic) [inline docs/src/use/configure/rules.md:287]: There's no `note`, we currently have only `warning`, `important`, or `tip`.

image

File: docs/src/use/configure/rules.md [hunk 18]
```diff
@@ -277,31 +305,47 @@ console.log('hello');
...
+    {
+        rules: {
+            "no-unused-expressions": "error"
+        }
+    },
     {
-      "files": ["*-test.js","*.spec.js"],
-      "rules": {
+      files: ["*-test.js","*.spec.js"],
+      rules: {
```
Comment (mdjermanovic) [inline docs/src/use/configure/rules.md:320]:
```suggestion
        files: ["*-test.js","*.spec.js"],
        rules: {
            "no-unused-expressions": "off"
        }
```

Just to fix the indentation.

File: docs/src/use/configure/rules.md [hunk 19]
```diff
@@ -277,31 +305,47 @@ console.log('hello');
...
+            "no-unused-expressions": "error"
+        }
+    },
+    {
+      files: ["*-test.js","*.spec.js"],
+      rules: {
+        "no-unused-expressions": "off"
+      }
+    }
+];
```
Comment (mdjermanovic) [inline docs/src/use/configure/rules.md:348]:
```suggestion
    }    
];
```

I think this is a copy & paste error, this doesn't seem relevant for this example.

File: docs/src/use/configure/index.md [hunk 20]
```diff
@@ -56,8 +51,6 @@ All of these options give you fine-grained control over how ESLint treats your c
 
 [**Ignore Files**](ignore)
 
-* [`ignorePatterns` in Config Files](./ignore#ignorepatterns-in-config-files)
-* [The `.eslintignore` File](./ignore#the-eslintignore-file)
-* [Using an Alternate File](./ignore#using-an-alternate-file)
-* [Using eslintIgnore in package.json](./ignore#using-eslintignore-in-packagejson)
+* [Ignoring Files](./ignore#ignoring-files)
+* [Unignoring Files](./ignore#ignoring-files)
```
Comment (mdjermanovic) [inline docs/src/use/configure/index.md:55]:
```suggestion
* [Unignoring Files](./ignore#unignoring-files)
```

File: docs/src/use/configure/index.md [hunk 21]
```diff
@@ -26,19 +25,15 @@ All of these options give you fine-grained control over how ESLint treats your c
...
-* [Personal Configuration Files](./configuration-files#personal-configuration-files-deprecated)
+* [Configuration File Format](./configuration-files#configuration-file)
+* [Configuration Objects](./configuration-files#configuration-objects)
+* [Configuring Shared Settings](./configuration-files#configuring-shared-settings)
+* [Configuration File Resolution](./configuration-files#configuration-file-resolution)
 
 [**Configure Language Options**](language-options)
 
-* [Specifying Environments](./language-options#specifying-environments)
+* [Specifying JavaScript Options](./language-options#specifying-parser-options)
```
Comment (mdjermanovic) [inline docs/src/use/configure/index.md:35]:
```suggestion
* [Specifying JavaScript Options](./language-options#specifying-javascript-options)
```

File: docs/src/use/configure/configuration-files.md [hunk 22]
```diff
@@ -1,451 +1,396 @@
 ---
-title: Configuration Files
+title: Configuration Files (New)
 eleventyNavigation:
     key: configuration files
     parent: configure
-    title: Configuration Files
-    order: 2
+    title: Configuration Files (New)
+    order: 1
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:7]: Is `(New)` intentional? Since this now becomes the default config format, perhaps there's no longer a need to mark it as "new".

File: docs/src/use/configure/configuration-files.md [hunk 23]
```diff
@@ -1,451 +1,396 @@
...
+    * `reportUnusedDisableDirectives` - A Boolean value indicating if unused disable and enable directives should be tracked and reported.
+* `processor` - Either an object containing `preprocess()` and `postprocess()` methods or a string indicating the name of a processor inside of a plugin (i.e., `"pluginName/processorName"`).
+* `plugins` - An object containing a name-value mapping of plugin names to plugin objects. When `files` is specified, these plugins are only available to the matching files.
+* `rules` - An object containing the configured rules. When `files` or `ignores` are specified, these rule configurations are only available to the matching files.
+* `settings` - An object containing name-value pairs of information that should be available to all rules.
+
+### Specifying `files` and `ignores`
+
+::: tip
+Patterns specified in `files` and `ignores` use [`minimatch`](https://www.npmjs.com/package/minimatch) syntax and are evaluated relative to the location of the `eslint.config.js` file. If using an alternate config file via the `--config` command line option, then all path are evaluated relative to the current working directory.
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:91]:
```suggestion
Patterns specified in `files` and `ignores` use [`minimatch`](https://www.npmjs.com/package/minimatch) syntax and are evaluated relative to the location of the `eslint.config.js` file. If using an alternate config file via the `--config` command line option, then all patterns are evaluated relative to the current working directory.
```

File: docs/src/use/configure/configuration-files.md [hunk 24]
```diff
@@ -1,451 +1,396 @@
...
+    * `sourceType` - The type of JavaScript source code. Possible values are `"script"` for traditional script files, `"module"` for ECMAScript modules (ESM), and `"commonjs"` for CommonJS files. (default: `"module"` for `.js` and `.mjs` files; `"commonjs"` for `.cjs` files)
+    * `globals` - An object specifying additional objects that should be added to the global scope during linting.
+    * `parser` - An object containing a `parse()` method or a `parseForESLint()` method. (default: [`espree`](https://github.com/eslint/espree))
+    * `parserOptions` - An object specifying additional options that are passed directly to the `parse()` or `parseForESLint()` method on the parser. The available options are parser-dependent.
+* `linterOptions` - An object containing settings related to the linting process.
+    * `noInlineConfig` - A Boolean value indicating if inline configuration is allowed.
+    * `reportUnusedDisableDirectives` - A Boolean value indicating if unused disable and enable directives should be tracked and reported.
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:82]:
```suggestion
    * `reportUnusedDisableDirectives` - A severity string indicating if and how unused disable and enable directives should be tracked and reported. For legacy compatibility, `true` is equivalent to `"warn"` and `false` is equivalent to `"off"`. (default: `"off"`).
```

This was updated in the meantime.

File: docs/src/use/configure/configuration-files.md [hunk 25]
```diff
@@ -1,451 +1,396 @@
...
 
-### Using `eslint:recommended`
-
-Using `"eslint:recommended"` in the `extends` property enables a subset of core rules that report common problems (these rules are identified with a checkmark (recommended) on the [rules page](../../rules/)).
-
-Here's an example of extending `eslint:recommended` and overriding some of the set configuration options:
+#### Reporting Unused Disable Directives
 
-Example of a configuration file in JavaScript format:
+Disable and enable directives such as `/*eslint-disable*/`, `/*eslint-enable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused disable directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:240]:
```suggestion
Disable and enable directives such as `/*eslint-disable*/`, `/*eslint-enable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused disable directives by setting the `reportUnusedDisableDirectives` option to a severity string, as in this example:
```

File: docs/src/use/configure/configuration-files.md [hunk 26]
```diff
@@ -1,451 +1,396 @@
...
-        "no-cond-assign": ["error", "always"],
-
-        // disable rules from base configurations
-         "for-direction": "off",
+// eslint.config.js
+export default [
+    {
+        files: ["**/*.js"],
+        linterOptions: {
+            reportUnusedDisableDirectives: true
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:248]:
```suggestion
            reportUnusedDisableDirectives: "error"
```

File: docs/src/use/configure/configuration-files.md [hunk 27]
```diff
@@ -1,451 +1,396 @@
...
-}
+];
 ```
 
-### Using a configuration from a plugin
-
-A [plugin](../../extend/plugins) is an npm package that can add various extensions to ESLint. A plugin can perform numerous functions, including but not limited to adding new rules and exporting [shareable configurations](../../extend/plugins#configs-in-plugins). Make sure the package has been installed in a directory where ESLint can require it.
-
-The `plugins` [property value](./plugins#configure-plugins) can omit the `eslint-plugin-` prefix of the package name.
+By default, unused disable and enable directives are reported as warnings. You can change this setting using the `--report-unused-disable-directives` command line option.
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:254]:
```suggestion
You can override this setting using the [`--report-unused-disable-directives`](../command-line-interface#--report-unused-disable-directives) or the [`--report-unused-disable-directives-severity`](../command-line-interface#--report-unused-disable-directives-severity) command line options.

For legacy compatibility, `true` is equivalent to `"warn"` and `false` is equivalent to `"off"`.
```

File: docs/src/use/configure/configuration-files.md [hunk 28]
```diff
@@ -1,451 +1,396 @@
...
-`~/.eslintrc.*` files behave similarly to regular configuration files, with some exceptions:
+```shell
+npx eslint --config some-other-file.js **/*.js
+```
 
-`~/.eslintrc.*` files load shareable configs and custom parsers from `~/node_modules/` – similarly to `require()` – in the user's home directory. Please note that it doesn't load global-installed packages.
+In this case, ESLint does not search for `eslint.config.js` and instead uses `some-other-file.js`.
 
-`~/.eslintrc.*` files load plugins from `$CWD/node_modules` by default in order to identify plugins uniquely. If you want to use plugins with `~/.eslintrc.*` files, plugins must be installed locally per project. Alternatively, you can use the [`--resolve-plugins-relative-to` CLI option](../command-line-interface#--resolve-plugins-relative-to) to change the location from which ESLint loads plugins.
+By default, all configuration objects are applied to all files ending with `.js`, `.cjs`, and `.mjs`. Sometimes, however, a more fine-grained application of configuration settings is necessary. For example, what if you'd like to apply a different configuration to different files in the same directory?
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:396]: This doesn't seem related to Configuration File Resolution.

File: docs/src/use/configure/configuration-files.md [hunk 29]
```diff
@@ -1,451 +1,396 @@
...
-│   ├── .eslintrc.json
-├── server
-│   ├── server.js
-│   ├── serverSpec.js
-├── .eslintrc.json
-```
+## Using a Shareable Configuration Package
 
-The config in `app/.eslintrc.json` defines the glob pattern `**/*Spec.js`. This pattern is relative to the base directory of `app/.eslintrc.json`. So, this pattern would match `app/lib/fooSpec.js` and `app/components/barSpec.js` but **NOT** `server/serverSpec.js`. If you defined the same pattern in the `.eslintrc.json` file within in the `project-root` folder, it would match all three of the `*Spec` files.
+A sharable configuration is an npm package that exports a configuration object or array. This package should be installed as a dependency in your project and then reference from inside of your `eslint.config.js` file. For example, to use a shareable configuration named `eslint-config-example`, your configuration file would look like this:
```
Comment (mdjermanovic) [inline docs/src/use/configure/configuration-files.md:344]:
```suggestion
A sharable configuration is an npm package that exports a configuration object or array. This package should be installed as a dependency in your project and then referenced from inside of your `eslint.config.js` file. For example, to use a shareable configuration named `eslint-config-example`, your configuration file would look like this:
```

File: docs/src/use/configure/configuration-files-new.md
Comment (nzakas) [reply docs/src/use/configure/configuration-files-new.md:1]: Oops, yes, good catch.

File: docs/src/use/configure/ignore.md [hunk 30]
```diff
@@ -156,7 +97,7 @@ foo.js
 
 This message occurs because ESLint is unsure if you wanted to actually lint the file or not. As the message indicates, you can use `--no-ignore` to omit using the ignore rules.
 
-Consider another scenario where you want to run ESLint on a specific dot-file or dot-folder, but have forgotten to specifically allow those files in your `.eslintignore` file. You would run something like this:
+Consider another scenario where you want to run ESLint on a specific dot-file or dot-folder, but have forgotten to specifically allow those files in your `eslint.config.js` file. You would run something like this:
```
[the remark this answers is not in the document]
Comment (nzakas) [reply docs/src/use/configure/ignore.md:100]: Good call, I'll remove it.

File: docs/src/use/configure/rules.md (hunk 17 above)
Comment (nzakas) [reply docs/src/use/configure/rules.md:286]: Thanks

File: docs/src/use/configure/configuration-files.md (hunk 28 above)
Comment (nzakas) [reply docs/src/use/configure/configuration-files.md:398]: Hmm yeah, looks like a copy-paste error.

File: docs/src/use/configure/configuration-files-new.md
Comment (mdjermanovic) [reply docs/src/use/configure/configuration-files-new.md:1]: Looks like the `configuration-files-new.md` file is still here:

image

Comment (mdjermanovic) [reply docs/src/use/configure/configuration-files-new.md:1]: It would be good to also update `order` fields in configuration documents. Since the rest of the changes look good, I'm going to merge this now to avoid future merge conflicts, and I'll submit another PR to update this.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!
